cohortextractor/cohortextractor.py

```diff
@@ -772,6 +774,21 @@ def main():
         except DummyDataValidationError as e:
             print(f"Dummy data error: {e}")
             sys.exit(1)
+        except database_exceptions() as e:
+            traceback.print_exc()
+            # Exit with specific exit codes to help identify known issues
+            if "Unexpected EOF from the server" in str(e):
+                logger.error(f"Intermittent database error: {e}")
+                sys.exit(151)
+            if "Invalid object name 'CodedEvent_SNOMED'" in str(e):
+                logger.error(
+                    "CodedEvent_SNOMED table is currently not available.\n"
+                    "This is likely due to regular database maintenance."
+                )
+                sys.exit(152)
+            logger.error(f"Database error: {e}")
```

Comment:
I've made these log here, but possibly job-runner should be responsible for looking for the specifc exit codes and logging there instead?  It checks for 137 (likely memory issues), but only in the local executor afaict.

Reply:
I think handling them here is correct.

In general, the sooner you handle an exception, the more context you have.

The job-runner only checks for 137 specifically because that very likely means the process was killed by the OOM, so it had no chance internally to catch and handle the error.